docs/userguides/transactions.md

```diff
@@ -0,0 +1,59 @@
+# Making Transactions
+
+Regardless of how you are using `ape`, you will likely be making transactions. There are various types of transactions
+you can make with `ape`. A simple example is deploying a contract.
+
+## Deployment
+
+Deploying a smart contract is a unique type of transaction where we don't necessarily care about the receipt as much
+as we care about the contract instance. That is why the return value from
+[the deploy method](../methoddocs/api.html?highlight=accountapi#ape.api.accounts.AccountAPI.deploy) is a
+[ContractInstance](../methoddocs/contracts.html?highlight=contractinstance#ape.contracts.base.ContractInstance).
+
+The following example demonstrates a simple deployment script:
+
+```python
+from ape import accounts, project
+
+
+def deploy():
+    account = accounts.load("MyAccount")
+    return account.deploy(project.MyContract)
+```
+
+## Dynamic Fee Transactions
+
+Before [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)), all transactions used a `gas_price`. After the London fork
+of Etheruem, the `gas_price` got broken up into two values, `max_fee` and `max_priority_fee`. The `ape` platform
+supports both types of transactions. By default, transactions use the dynamic fee model. Making contract calls without
+specifying any additional `kwargs` will use a dynamic-fee transaction.
+
+Calling certain methods on a deployed-contract is one way to transact.
+
+```python
+contract = deploy()  # Example from above, that returns a contract instance.
+contract.fundMyContract(value="1 gwei")  # Assuming there is a method named 'fundMyContract' on MyContract.
+```
+
+In the example above, the call to `fundMyContract()` invokes a dynamic-fee transaction. To have more control of the 
+fee-values, you can specify the `max_fee`, the `max_priority_fee`, or both.
+
+```python
+contract.fundMyContract(value="1 gwei", max_priority_fee="50 gwei", max_fee="100 gwei")
+```
+
+## Static Fee Transactions
+
+One way to use a static-fee transaction is by specifying the `gas_price` as a key-value argument:
+
+```python
+contract.fundMyContract(value="1 gwei", gas_price="100 gwei")
+```
+
+Another way to use a static-fee transaction (without having to provide `gas_price`) is to set the key-value
+argument `type` equal to `0x00`.
+
+```python
+contract.fundMyContract(value="1 gwei", type="0x0")
+```
+
```

Comment:
Add a note that `max_priority_fee` cannot exceed `max_fee`, and that `base_fee = max_fee - max_priority_fee`, which much match network conditions in order to be mined.

Lastly, what happens if `max_priority_fee` is omitted or `max_fee` is omitted in terms of the value of the unspecified fee?

I would think that specifying `max_priority_fee` means `max_fee = ProviderAPI.base_fee + max_priority_fee`, and that specifying `max_fee` without `max_priority_fee` means `max_priority_fee` will be some average over the last N blocks of that value, and `base_fee` is set to `max_fee - max_priority_fee` (and might take longer to confirm based on the current value of `base_fee`, which we can create a warning about)

Reply:
Also I should specify that we have to do this lmao

Reply:
I think your expectations of what happens is correct. Here is where the defaults get set: https://github.com/ApeWorX/ape/blob/main/src/ape/api/accounts.py#L109-L115

* If you omit the `max_priority_fee` and and include the `max_fee`, the `max_priority_fee` just gets set from the provider.

* If you omit the `max_fee` and include the `max_priority_fee`, the `max_fee` still just becomes the `base_fee` plus the `max_priority_fee`.

As far as I know, there is no way to set the base fee. I don't know how `priority_fee` gets calculated by the provider.

Reply:
> As far as I know, there is no way to set the base fee.

That is correct, it is essentially a "protocol parameter" set by the previous block and how full/empty it is.

> And I don't know how `priority_fee` gets calculated by the provider.

The formula `max_fee = base_fee + priority_fee` is probably used.